scripts/lib/CIME/namelist.py

```diff
@@ -884,25 +1000,59 @@ def get_value(self, variable_name):
         else:
             return [u'']
 
-    def set_variable_value(self, group_name, variable_name, value):
+    def set_variable_value(self, group_name, variable_name, value, var_size=1):
         """Set the value of the specified variable.
 
         >>> x = parse(text='&foo bar=1 /')
+        >>> x.get_variable_value('foo', 'bar')
+        [u'1']
+        >>> x.set_variable_value('foo', 'bar(2)', [u'3'], var_size=4)
+        >>> x.get_variable_value('foo', 'bar')
+        [u'1', u'3']
         >>> x.set_variable_value('foo', 'bar', [u'2'])
+        >>> x.get_variable_value('foo', 'bar')
+        [u'2', u'3']
         >>> x.set_variable_value('foo', 'bazz', [u'3'])
         >>> x.set_variable_value('Brack', 'baR', [u'4'])
-        >>> x.get_variable_value('foo', 'bar')
-        [u'2']
         >>> x.get_variable_value('foo', 'bazz')
         [u'3']
         >>> x.get_variable_value('brack', 'bar')
         [u'4']
+        >>> x.set_variable_value('foo', 'red(2:6:2)', [u'2', u'4', u'6'], var_size=12)
+        >>> x.get_variable_value('foo', 'red')
+        ['', u'2', '', u'4', '', u'6']
         """
         group_name = group_name.lower()
-        variable_name = variable_name.lower()
+
+#        print "BEFORE name %s value %s"%(variable_name, value)
+        minindex, maxindex, step = get_fortran_variable_indices(variable_name, var_size)
+        variable_name = get_fortran_name_only(variable_name.lower())
+
+        expect(minindex > 0, "CIME indices < 1 not supported in CIME interface to fortran namelists minindex=%s"%minindex)
```

Comment:
But the expect logic is correct here  - if in user_nl you have dtlimit(6) then minindex=maxindex=6,step=1 .